pyretailscience/plots/line.py

```diff
@@ -0,0 +1,317 @@
+
+"""
+
+import logging
+
+import pandas as pd
+from matplotlib.axes import Axes, SubplotBase
+
+import pyretailscience.style.graph_utils as gu
+from pyretailscience.style.graph_utils import GraphStyles
+from pyretailscience.style.tailwind import COLORS
+
+logging.basicConfig(format="%(message)s", level=logging.INFO)
+
+
+def _check_datetime_column(df: pd.DataFrame, x_col: str) -> None:
+    """Checks if the x_col is a datetime or convertible to datetime.
+
+    Issues a warning if the column is datetime-like, recommending
+    the use of a time-based plot.
+
+    Args:
+        df (pd.DataFrame): The dataframe containing the column to check.
+        x_col (str): The column to check for datetime-like values.
+    """
+    if pd.api.types.is_datetime64_any_dtype(df[x_col]):
+        logging.warning(
+            "The column '%s' is a datetime column. Consider using the 'time_plot' function for time-based plots.",
+            x_col,
+        )
+    else:
+        try:
+            pd.to_datetime(df[x_col])
+            logging.warning(
+                "The column '%s' can be converted to datetime. Consider using the 'time_plot' module for time-based plots.",
+                x_col,
+            )
+        except (ValueError, TypeError):
+            pass
+
+
+def plot(
+        df: pd.DataFrame,
+        value_col: str | list[str],
+        group_col: str | None = None,
+        x_col: str | None = None,
+        title: str | None = None,
+        x_label: str | None = None,
+        y_label: str | None = None,
+        legend_title: str | None = None,
+        ax: Axes | None = None,
+        source_text: str | None = None,
+        move_legend_outside: bool = False,
+        **kwargs: dict[str, any],
+) -> SubplotBase:
+    """Plots the `value_col` over the specified `x_col` or index, creating a separate line for each unique value in `group_col`.
+
+    Args:
+        df (pd.DataFrame): The dataframe to plot.
+        value_col (str or list of str): The column(s) to plot.
+        group_col (str, optional): The column used to define different lines.
+        x_col (str, optional): The column to be used as the x-axis. If None, the index is used.
+        title (str, optional): The title of the plot.
+        x_label (str, optional): The x-axis label.
+        y_label (str, optional): The y-axis label.
+        legend_title (str, optional): The title of the legend.
+        ax (Axes, optional): Matplotlib axes object to plot on.
+        source_text (str, optional): The source text to add to the plot.
+        move_legend_outside (bool, optional): Move the legend outside the plot.
+        **kwargs: Additional keyword arguments for Pandas' `plot` function.
+
+    Returns:
+        SubplotBase: The matplotlib axes object.
+    """
+    if x_col is not None:
+        _check_datetime_column(df, x_col)
+
+    if group_col is not None:
+        unique_groups = df[group_col].unique()
+        colors = [
```

Comment:
Can you create this as a listed_cmap. See [here ](https://github.com/Data-Simply/pyretailscience/blob/main/pyretailscience/style/tailwind.py#L375 )for where the base ones are called. I'd just create a "rainbow" cmap and have a function that builds it.